### PR TITLE
Bug Fix - Update Docker Exec Command for Persistent HPCX Environment

### PR DESCRIPTION
**Pull request by RyoYang**
Add 10-hpcx.sh to /etc/profile.d
Update the Docker exec command to ensure a persistent HPCX environment.

### CODE REVIEW

File: dockerfile/cuda12.2.dockerfile [hunk 1]
```diff
@@ -131,7 +131,7 @@ ENV PATH="${PATH}" \
 RUN echo PATH="$PATH" > /etc/environment && \
     echo LD_LIBRARY_PATH="$LD_LIBRARY_PATH" >> /etc/environment && \
     echo SB_MICRO_PATH="$SB_MICRO_PATH" >> /etc/environment && \
-    echo "source /opt/hpcx/hpcx-init.sh && hpcx_load" >> /etc/bash.bashrc
+    echo "source /opt/hpcx/hpcx-init.sh && hpcx_load" >> /etc/profile.d/10-hpcx.sh
```
Comment (abuccts) [inline dockerfile/cuda12.2.dockerfile:134]:
```suggestion
    echo "source /opt/hpcx/hpcx-init.sh && hpcx_load" | tee -a /etc/bash.bashrc >> /etc/profile.d/10-hpcx.sh
```

Comment (RyoYang) [reply dockerfile/cuda12.2.dockerfile]: Updated

**Review by abuccts — APPROVED** [review]

**Review by cp5555 — APPROVED** [review]

**Pull request merged**